dask_searchcv/tests/test_adaptive.py

```diff
@@ -0,0 +1,187 @@
+from dask_searchcv.adaptive import _top_k, Hyperband
+from sklearn.linear_model import SGDClassifier
+from dask_ml.datasets import make_classification
+import numpy as np
+import dask.array as da
+from dask.distributed import Client
+import pandas as pd
+import pytest
+import distributed
+from pprint import pprint
+import scipy.stats as stats
+import random
+from sklearn.linear_model import Lasso
+from distributed.utils_test import cluster, loop, gen_cluster
+import time
+from dask_ml.wrappers import Incremental
+
+
+class ConstantFunction:
+    def _fn(self):
+        return self.value
+
+    def get_params(self, deep=None, **kwargs):
+        return {k: getattr(self, k) for k, v in kwargs.items()}
+
+    def set_params(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        return self
+
+    def partial_fit(self, *args, **kwargs):
+        pass
+
+    def score(self, *args, **kwargs):
+        return self._fn()
+
+
+def _get_client():
+    try:
+        return distributed.get_client()
+    except:
+        return Client()
+
+
+def _with_client(fn):
```

Comment:
See http://distributed.readthedocs.io/en/latest/develop.html#writing-tests

Reply:
I recommend going with standard testing style, both because it's well tested (and things can get hairy) and because uniformity across the project is useful to enable review.  

That being said, I like how this makes things less wordy.  If ever feel like a simple-yet-productive task, you might be interested in https://github.com/dask/distributed/issues/1966